src/Parser.js

```diff
@@ -165,7 +165,8 @@ class Parser {
             if (breakOnInfix && functions[lex.text] && functions[lex.text].infix) {
                 break;
             }
-            const atom = this.parseAtom();
+            const optional = breakOnTokenText === "]";
+            const atom = this.parseAtom(optional);
```

Comment:
I haven't thought about this deeply, but is there a reason to just pass in `]` and not just `breakOnTokenText` in general?  I think the latter might be useful for more general macro support in the future... Maybe worth reviewing what other values this currently takes on in the code, if any.

Reply:
Good idea.  I'll look into this evening.